Feat : 입출금 이자 구현, 이자지급 Enum 추가

Wonhee0221 opened the pull request:
##################
스케줄러 사용 매달 15일 마다 실행

### PR Type

- [x] 기능 개발 (Feature)
- [ ] 코드 스타일 변경 (Code style update) (formatting, local variables)
- [ ] 화면 작업 (Style, CSS)
- [ ] 리팩토링 (no functional changes, no api changes)
- [ ] 버그수정 (Bugfix)
- [ ] 빌드, 배포 관련 변경 (Build, Deploy)
- [ ] 문서 작업 (Docs)
- [ ] 그 외 (ETC)

## 작업 내용

- 스케줄러 사용 매달 15일 마다 실행
- 이자지급 Enum 추가

## 참고 사항, 첨부 자료 등

Wonhee0221 committed:
[feat] 입출금 이자 구현, 이자지급 Enum 추가
##################
스케줄러 사용 매달 15일 마다 실행

soyeonvv commented on src/main/java/com/hana/bankai/global/scheduler/BankaiScheduler.java:
```diff
@@ -20,4 +23,7 @@ public void resetDailyUserTrsfLimit() {
     @Scheduled(cron = "0 0 12 * * ?") // 매일 낮 12시에 실행
     public void autoTransfer() { autoTransferService.autoTransfer(); }
+
+    @Scheduled(cron = "0 0 0 15 * ?") // 매달 15일에 실행
```
보통 이자가 자정에 들어오지 않으니까 낮 12시로 하는 거 어떤가요??

Wonhee0221 committed:
[feat] 최소, 최대 납입 금액 제한 기능 구현 및 이자 지급 스케줄러 수정

##################
AccountService
validatePostTransferAmount : 납입시 상품의 최대 납입 금액을 넘어 가지 않게 확인.
validateProductAmount : 최소 납입, 최내 납입 금액확인.
ErrorCode - 납입 에러 추가

soyeonvv commented:
LGTM!
